compose/config/types.py

```diff
@@ -457,3 +459,30 @@ def normalize_port_dict(port):
         external_ip=port.get('external_ip', ''),
         has_ext_ip=(':' if port.get('external_ip') else ''),
     )
+
+
+class SecurityOpt(namedtuple('_SecurityOpt', 'value src_file')):
+    @classmethod
+    def parse(cls, value):
+        # based on https://github.com/docker/cli/blob/9de1b162f/cli/command/container/opts.go#L673-L697
+        con = value.split('=', 2)
+        if len(con) == 1 and con[0] != 'no-new-privileges':
+            if ':' not in value:
+                raise ConfigurationError('Invalid security_opt: {}'.format(value))
+            con = value.split(':', 2)
+
+        if con[0] == 'seccomp' and con[1] != 'unconfined':
```

Comment:
Previously, it was possible to pass a seccomp profile by embedding the JSON-encoded profile into the docker-copmose file. This won't work anymore, because every value other than `unconfined` is now interpreted as a path. Maybe add `and not con[1].startswith('{')` here to leave alone values that look like JSON.